cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataHttpHandler.java

```diff
@@ -222,7 +223,7 @@ public void addAppProperties(HttpRequest request, HttpResponder responder,
                                @PathParam("namespace-id") String namespaceId,
                                @PathParam("app-id") String appId) throws BadRequestException, NotFoundException {
     Id.Application app = Id.Application.from(namespaceId, appId);
```

Comment:
hmmm, why not use the new ApplicationId, etc classes here instead of doing a ``toEntityId`` on the next line? Applies to all methods in this class.